tests/test_wavex.py

```diff
@@ -0,0 +1,288 @@
+        == wavex_model.components["WaveX"].wavex_delay(toas, 0.0 * u.s)
+    )
+
+
+def test_multiple_wavex_broadcast_frozens():
+    # Check that when a single False is given for frozens, it gets broadcast to all the sine and cosine amplitudes
+    model = get_model(StringIO(par2))
+    indices = model.components["WaveX"].add_wavex_components(
+        [0.2, 0.3],
+        indices=[2, 3],
+        wxsins=[2, 3],
+        wxcoses=[2, 3],
+        frozens=False,
+    )
+    for index in indices:
+        assert getattr(model, f"WXSIN_{index:04d}").frozen == False
+        assert getattr(model, f"WXCOS_{index:04d}").frozen == False
+
+
+def test_multiple_wavex_wrong_cos_amps():
+    # Check that code breaks when adding an extra cosine amplitude than there are frequencies, indices, and sine amplitudes for
+    model = get_model(StringIO(par2))
+    with pytest.raises(ValueError):
+        indices = model.components["WaveX"].add_wavex_components(
+            [0.2, 0.3], indices=[2, 3], wxsins=[2, 3], wxcoses=[2, 3, 4]
+        )
+
+
+def test_multiple_wavex_wrong_sin_amps():
+    # Check that code breaks when adding an extra sine amplitude than there are frequencies, indices, and cosine amplitudes for
+    model = get_model(StringIO(par2))
+    with pytest.raises(ValueError):
+        indices = model.components["WaveX"].add_wavex_components(
+            [0.2, 0.3], indices=[2, 3], wxsins=[2, 3, 4], wxcoses=[2, 3]
+        )
+
+
+def test_multiple_wavex_wrong_freqs():
+    # Check that code breaks when not adding enough frequencies for the number of indices, sine amps, and cosine amps given
+    model = get_model(StringIO(par2))
+    with pytest.raises(ValueError):
+        indices = model.components["WaveX"].add_wavex_components(
+            [0.2, 0.3], indices=[2, 3, 4], wxsins=[2, 3, 4], wxcoses=[2, 3, 4]
+        )
+
+
+def test_multiple_wavex_wrong_frozens():
+    # Check that adding to many elements to frozens breaks code
+    model = get_model(StringIO(par2))
+    with pytest.raises(ValueError):
+        indices = model.components["WaveX"].add_wavex_components(
+            [0.2, 0.3],
+            indices=[2, 3],
+            wxsins=[2, 3],
+            wxcoses=[2, 3],
+            frozens=[False, False, False],
+        )
+
+
+def test_multiple_wavex_explicit_indices():
+    # Check that adding specific indices is done correctly
+    model = get_model(StringIO(par2))
+    indices = model.components["WaveX"].add_wavex_components(
+        [0.2, 0.3], indices=[3, 4], wxsins=[2, 3], wxcoses=[2, 3]
+    )
+    assert np.all(np.array(indices) == np.array([3, 4]))
+
+
+def test_multiple_wavex_explicit_indices_duplicate():
+    # Check that adding a duplicate index fails
+    model = get_model(StringIO(par2))
+    with pytest.raises(ValueError):
+        indices = model.components["WaveX"].add_wavex_components(
+            [0.2, 0.3], indices=[1, 3], wxsins=[2, 3], wxcoses=[2, 3]
+        )
```

Comment:
If you add the Wave <-> WaveX functions, then you can also do a roundtrip test where you start with one, translate to the other, and go back.  That should be the same as what you started with.

Reply:
Added the roundtrip test